schedule/scheduler.go

```diff
@@ -290,6 +291,40 @@ func (s *Scheduler) StopJob(id JobID) error {
 	return nil
 }
 
+func (s *Scheduler) WaitJob(id JobID) (*JobListResult, error) {
+	s.jobsMu.Lock()
+	job, ok := s.jobs[id]
+	if !ok {
+		return nil, xerrors.Errorf("wait job ID: %d not found", id)
```

Comment:
Unlock `jobsMu` before return. Can just move the existing `Unlock` to just before the `if`

Reply:
whoops, done.